src/store/mod.rs

```diff
@@ -1,11 +1,13 @@
 use crate::model::Pod;
-use std::{collections::BTreeMap, error::Error};
+use anyhow::Result;
+use std::collections::BTreeMap;
 
 pub trait Store {
-    fn save_pod(&self, pod: &Pod) -> Result<(), Box<dyn Error>>;
-    fn load_pod(&self, name: &str, version: &str) -> Result<Pod, Box<dyn Error>>;
-    fn list_pod(&self) -> Result<BTreeMap<String, Vec<String>>, Box<dyn Error>>;
-    fn delete_pod(&self, name: &str, version: &str) -> Result<(), Box<dyn Error>>;
+    fn save_pod(&mut self, pod: &Pod) -> Result<()>;
+    fn load_pod(&mut self, name: &str, version: &str) -> Result<Pod>;
+    fn list_pod(&mut self) -> Result<&BTreeMap<String, String>>;
+    fn delete_pod(&mut self, name: &str, version: &str) -> Result<()>;
```

Comment:
Assuming `delete_annotation` below is the explicit delete of 1 annotation, would this then be an explicit delete of a whole pod i.e. the pod spec directory + all annotation references?

If so, then it might be better to change the input signature to just require a `hash` argument.

Reply:
delete annotation, will only delete annotation and not the underlying object. Going off of what I remember we discussed last week. Delete_pod will wipe the entire item along with all annotations